HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+*/
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include <opencv2/core/core.hpp>
+
+#include <cmath>
+#include <iostream>
+
+#define _USE_MATH_DEFINES
+
+namespace TransformUtils
+{
+    enum Rotation
+    {
+        XYZ_INTRINSIC,
+        XYZ_EXTRINSIC,
+        ZYX_INTRINSIC,
+        ZYX_EXTRINSIC
+    };
+
+    struct Rotation_s
+    {
+        Rotation rot;
+        std::string short_desc;
+        std::string rpy_name;
+    };
+
+    cv::Mat eigen2cv(const Eigen::MatrixXd &eigenMat)
+    {
+        cv::Mat cvMat(static_cast<int>(eigenMat.rows()), static_cast<int>(eigenMat.cols()), CV_64FC1, cv::Scalar(0));
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                cvMat.at<double>(i, j) = eigenMat(i, j);
+            }
+        }
+
+        return cvMat;
+    }
+
+    Eigen::MatrixXd cv2eigen(const cv::Mat &cvMat)
+    {
+        if(cvMat.dims > 2)
+        {
+            throw "Invalid matrix dimensions. Expected 2D.";
+        }
+
+        Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                eigenMat(i, j) = cvMat.at<double>(i, j);
+            }
+        }
+
+        return eigenMat;
+    }
+
+    Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+    {
+        Eigen::Vector4d reorderedAxisAngle;
+
+        for(int i = 0; i < 3; i++)
+            reorderedAxisAngle[i] = axisAngle.axis()[i];
+        reorderedAxisAngle[3] = axisAngle.angle();
+
+        return reorderedAxisAngle;
+    }
+
+    /** Converts Rotation Matrix to RollPitchYaw.
+     * The convention we use here is that Roll is a rotation about x-axis,
+     * Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+     * @param rotationMatrix Rotation Matrix to convert to RPY
+     * @param rotation Enum to indicate which convention to follow for this conversion
+     * @return rollPitchYaw array of angles, in radians
+     */
+    Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d rotationMatrix,
+                              const TransformUtils::Rotation rotation = ZYX_INTRINSIC)
+    {
+        Eigen::Array3d rollPitchYaw;
+        switch(rotation)
+        {
+            case TransformUtils::Rotation::XYZ_INTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+            case TransformUtils::Rotation::XYZ_EXTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+            case TransformUtils::Rotation::ZYX_INTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+            case TransformUtils::Rotation::ZYX_EXTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+            default: throw "Invalid rotation";
+        }
+        // The returned array should be ordered by Ox, Oy and Oz. Where Oa is the rotation about axis a.
```

Comment:
Notpick - replace dot for a comma?

The returned array should be ordered by Ox, Oy and Oz, where Oa is the rotation about axis a.